src/Database/LSMTree/Internal/PageAcc1.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+-- | Create a single value page
+module Database.LSMTree.Internal.PageAcc1 (
+    singletonPage,
+) where
+
+import           Control.Monad.ST.Strict (ST, runST)
+import qualified Data.Primitive as P
+import qualified Data.Vector.Primitive as PV
+import           Data.Word (Word16, Word32, Word64)
+import           Database.LSMTree.Internal.BlobRef (BlobSpan (..))
+import           Database.LSMTree.Internal.Entry (Entry (..))
+import           Database.LSMTree.Internal.RawPage
+import           Database.LSMTree.Internal.Serialise
+import           Database.LSMTree.Internal.Serialise.RawBytes (RawBytes (..))
+
+pageSize :: Int
+pageSize = 4096
+{-# INLINE pageSize #-}
+
+-- | Create a singleton page, also returning the overflow value bytes.
+singletonPage
+    :: SerialisedKey
+    -> Entry SerialisedValue BlobSpan
+    -> (RawPage, RawBytes)
+singletonPage k (Insert v) = runST $ do
+    -- allocate bytearray
+    ba <- P.newByteArray pageSize :: ST s (P.MutableByteArray s)
+    P.fillByteArray ba 0 pageSize 0
```

Comment:
Makes the code more obviously correct, good. You could leave a TODO to try optimising it away later (during benchmarking & optimisation phase), if you think it makes sense.

Reply:
I don't think it makes. `memset` is cheap.